In [2]:
import pandas as pd
import numpy as np
from scipy import stats

# Question 1

**α==0.05 (95% Confidence level)
Ho : There is no significant difference in diameter of cutlets in two units. i.e. µ1=µ2
H1 : There is significant difference in diameter of cutlets in two units. i.e. µ1=!µ2**

# 2 samples hence 2 tailed test is aplicable

In [4]:
cutlet=pd.read_csv("Cutlets.csv")

In [6]:
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [7]:
unitA=pd.Series(cutlet.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [8]:
unitB=pd.Series(cutlet.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [9]:
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [10]:
p_value[1]

0.4722394724599501

In [11]:
np.round(p_value[1],2)

0.47

In [15]:
print('p-value==0.47>α' )

p-value==0.47>α


# p-value==0.47>α , Hence Ho accepted

# There is no significant difference in diameter of cutlets in two units.

              *************************************************************************************

In [21]:
from scipy.stats import norm

# Question 2 (LabTat)

**α==0.05(95% confidence level)
Ho= M1,M2,M3 and M4 are same(normal); M: mean
H1= M1,M2,M3 and M4 are not same(normal); M: mean **  

In [24]:
labtat=pd.read_csv("LabTat.csv")

In [25]:
labtat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [26]:
stats.shapiro(labtat["Laboratory 1"])

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)

p-value for laboratory 1 == 0.55>α

In [27]:
stats.shapiro(labtat["Laboratory 2"])

ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)

p-value for laboratory 2 == 0.86>α

In [28]:
stats.shapiro(labtat["Laboratory 3"])

ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)

p-value for laboratory 3 == 0.42>α

In [29]:
stats.shapiro(labtat["Laboratory 4"])

ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)

p-value for laboratory 4 == 0.66>α

# Hence Ho is accepted.Thus M1,M2,M3 and M4 are normal

# varience


HO: Variance of 4 libraries are same.
Ha: Variance of 4 libraries not are same.    

In [30]:
stats.levene(labtat['Laboratory 1'],labtat['Laboratory 2'],labtat['Laboratory 3'],labtat['Laboratory 4'])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

p-value of variance == 0.051> α

# Ho is accepted. Hence we perform one way anocova test 

HO : mean TAT of 4 laboratories are equal.
H1 : mean TAT of 4 laboratories are not equal.    
    

In [31]:
stats.f_oneway(labtat['Laboratory 1'],labtat['Laboratory 2'],labtat['Laboratory 3'],labtat['Laboratory 4'])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [33]:
p_value=print('pvalue=2.1156708949992414e-57')

pvalue=2.1156708949992414e-57


p value is 2e-16< α

# H1 is accepted.mean TAT of 4 laboratories are not equal.

                   ***************************************************************************************

# Question 3

α == 0.05(95% confidence level)
Ho:male-female buyer rations are similar across regions.
H1:male-female buyer rations are not similar across regions.    

# Chi-square test

In [36]:
br=pd.read_csv("BuyerRatio.csv")

In [38]:
br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [40]:
stats.chi2_contingency([br["East"],br["West"],br["North"],br["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

p value for chi-square test == 0.66 > α

# Ho is accepted.The male-female buyer rations are similar across regions.      

                        *********************************************************************************

# Question 4

In [ ]:
α == 0.05(95% confidence level)

HO:the defective % dose not varies by centre. 
H1:the defective % dose varies by centre.     

In [43]:
df=pd.read_csv("Costomer+OrderForm.csv")

In [44]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [49]:
stats.chi2_contingency([df["Phillippines"].value_counts(),df["Indonesia"].value_counts(),df["Malta"].value_counts(),df["India"].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

p-value of Chi-square test is 0.277> α

# Ho is accepted.the defective % dose not varies by centre. 

In [ ]:
                 ***************************************************************************************